designer/server/src/main/scala/pl/touk/nussknacker/ui/additionalconfig/AdditionalUIConfigProvider.scala

```diff
@@ -0,0 +1,78 @@
+package pl.touk.nussknacker.ui.additionalconfig
+
+import pl.touk.nussknacker.engine.api.component._
+import pl.touk.nussknacker.engine.api.definition.FixedExpressionValue
+import pl.touk.nussknacker.engine.compile.nodecompilation.ValueEditorValidator
+import pl.touk.nussknacker.engine.graph.node.{ParameterValueCompileTimeValidation, ValueInputWithFixedValues}
+
+/**
+ * Trait allowing the provision of UI configuration for components and scenario properties, without requiring a model reload.
+ *
+ * TODO: The current implementation allows providing configs only for standard components - meaning that base components aren't handled.
+ * TODO: Currently the value of 'valueCompileTimeValidation' has no effect, it'll be supported in the future but is included now to keep the API stable.
+ *       Validating and extracting a parameter's validators requires access to expressionCompiler and validationContext,
+ *       this is currently hard to achieve where AdditionalUIConfigProvider is used (UIProcessObjectsFactory),
+ *       after refactoring it to be used at the level of ModelData or so it should be easy, and support for 'valueCompileTimeValidation' will be possible
+ */
+trait AdditionalUIConfigProvider extends Serializable {
```

Comment:
I think that it needs to be in some api module :) `component-api` or `exetensions-api`. How otherwise you would provide your own implementation of plugin?